File: house/controllers/get_controller.py
from house.controllers.basic_controller import *

class GetController(BasicController):
    """docstring for LoadController."""
    def __init__(self, args):
        super().__init__(args)

    def draw(self):
        dirs = self.args[0].split('/')
        levels = 1 + len(dirs)
        commands = []

        commands.append('cd src')
        for direc in dirs:
            commands.append('cd ' + direc)
        commands.append('git clone https://{0}.git'.format(self.args[0]))
        for _ in range(levels):
            commands.append('cd ..')

        self.execute(commands)

File: house/controllers/get_controller.lua
local basic_controller = require "house/controllers/basic_controller"
local get_controller = { }

get_controller.new = function(args)
    local self = basic_controller.new(args)

    self.draw = function()
        local dirs = util.mysplit(self.args[1], "/")
        local levels = #dirs
        local commands = { }

        table.insert(commands, "cd src")
        for _, d in ipairs(dirs) do
            table.insert(commands, "cd " .. d)
        end
        table.insert(commands, "git clone https://" .. self.args[1] .. ".git")
        for i = 1, levels do
            table.insert(commands, "cd ..")
        end

        self.execute(commands)
    end

    return self
end

return get_controller
